src/download-artifact.ts

```diff
@@ -8,14 +9,20 @@ async function run(): Promise<void> {
     const name = core.getInput(Inputs.Name, {required: false})
     const path = core.getInput(Inputs.Path, {required: false})
 
+    // resolve tilde expansion
+    const resolvedPath = resolve(path.replace('~', os.homedir))
```

Comment:
If the user doesn't specify a path, it will be an empty string. Calling `resolve` on an empty string returns the current working directory

Reply:
Would it be possible to use this?
https://github.com/actions/toolkit/blob/1cc56db0ff126f4d65aeb83798852e02a2c180c3/packages/glob/README.md#tilde-expansion

Reply:
Nope, we can't use `actions/glob` to just resolve tilde expansions. It fundamentally servers a different purpose.

Internally I hunted down though where glob resolves tilde expansion: https://github.com/actions/toolkit/blob/1cc56db0ff126f4d65aeb83798852e02a2c180c3/packages/glob/src/internal-pattern.ts#L213

Reply:
I modified the logic a little bit so that it only replaces with `os.homedir` if the path starts with a tilde. In all other scenarios it should not be necessary

Reply:
> It fundamentally servers a different purpose.

Can you please elaborate?  I thought this is just expanding `~` to user home dir.  If `upload-artifact` is using `glob`, why would `download-artifact` not be able to use that?